src/groups/special_euclidean.jl

```diff
@@ -25,34 +25,45 @@ $\mathrm{T}(n) × \mathrm{SO}(n)$. For group-specific functions, they may also b
 represented as affine matrices with size `(n + 1, n + 1)` (see [`affine_matrix`](@ref)), for
 which the group operation is [`MultiplicationOperation`](@ref).
 """
-const SpecialEuclidean{N} = SemidirectProductGroup{
+const SpecialEuclidean{T} = SemidirectProductGroup{
     ℝ,
-    TranslationGroup{Tuple{N},ℝ},
-    SpecialOrthogonal{N},
-    RotationAction{TranslationGroup{Tuple{N},ℝ},SpecialOrthogonal{N},LeftForwardAction},
+    TranslationGroup{T,ℝ},
+    SpecialOrthogonal{T},
+    RotationAction{TranslationGroup{T,ℝ},SpecialOrthogonal{T},LeftForwardAction},
 }
 
 const SpecialEuclideanManifold{N} =
-    ProductManifold{ℝ,Tuple{TranslationGroup{Tuple{N},ℝ},SpecialOrthogonal{N}}}
+    ProductManifold{ℝ,Tuple{TranslationGroup{N,ℝ},SpecialOrthogonal{N}}}
 
-function SpecialEuclidean(n)
-    Tn = TranslationGroup(n)
-    SOn = SpecialOrthogonal(n)
+function SpecialEuclidean(n; parameter::Symbol=:field)
+    Tn = TranslationGroup(n; parameter=parameter)
+    SOn = SpecialOrthogonal(n; parameter=parameter)
     A = RotationAction(Tn, SOn)
     return SemidirectProductGroup(Tn, SOn, A)
 end
 
 const SpecialEuclideanOperation{N} = SemidirectProductOperation{
-    RotationAction{TranslationGroup{Tuple{N},ℝ},SpecialOrthogonal{N},LeftForwardAction},
+    RotationAction{TranslationGroup{N,ℝ},SpecialOrthogonal{N},LeftForwardAction},
 }
 const SpecialEuclideanIdentity{N} = Identity{SpecialEuclideanOperation{N}}
 
-Base.show(io::IO, ::SpecialEuclidean{n}) where {n} = print(io, "SpecialEuclidean($(n))")
+function Base.show(io::IO, ::SpecialEuclidean{TypeParameter{Tuple{n}}}) where {n}
+    return print(io, "SpecialEuclidean($(n); parameter=:type)")
+end
+function Base.show(io::IO, G::SpecialEuclidean{Tuple{Int}})
+    n = get_n(G)
+    return print(io, "SpecialEuclidean($(n))")
+end
 
 @inline function active_traits(f, M::SpecialEuclidean, args...)
     return merge_traits(IsGroupManifold(M.op), IsExplicitDecorator())
 end
 
+get_n(::SpecialEuclidean{TypeParameter{Tuple{N}}}) where {N} = N
```

Comment:
here again the same – I think the generic get_parameter should be fine as well?

Reply:
This is one place where `get_parameter` is less clear, as `SpecialEuclidean` requires a lot of unwrapping: `get_parameter(M.manifold.manifolds[1].manifold.size)[1]`. Can we have a helper function that does that?

Reply:
We could indeed just define `get_parameter(::SpecialEuclidean{TypeParameter{Tuple{N}}}) where {N} = N` or would that lead to ambiguities since `SpecialEuclidean` is some strange type?

Reply:
For me the problem is that similar to “magic numbers” the `n` in the function name is a “magic name”.
We have a few ones we document and use a lot (M, p, q, X, Y, and such) but not `n` in that general sense yet, that is my argument against the function name.

Reply:
`get_parameter` was specifically written to unwrap `TypeParameter`. Adding special behavior for manifolds may be problematic in case a manifold is used as a (type) parameter. I'm open to calling this helper function something different than `get_n` but don't have an idea for a good name.

Reply:
it is, after all, still getting the parameter, what the function does, though :D

Would `_get_parameter` be ok? That indicates that this is a local function, here even meant in the sense only local to special euclidean even?

Reply:
`_get_parameter` is definitely fine. I just don't want `get_parameter` to ever return a parameter of a parameter instead of a parameter (a bit like `1\in \{1\}` but `1 \notin \{\{1\}\}`)

Reply:
I see your point and agree. But then let's use `_get_parameter` here.